### ISSUE TITLE
"input" should not be a reserved parameter

### ISSUE THREAD

**Issue by ben-manes** [h908714124/auto-builder#18]
The constructor uses the parameter name `input` and does not qualify fields using `this`. That means the field `input` is not allowed.

```java
@AutoValue @AutoBuilder
public static abstract class WatermarkSettings {
  public abstract int fontSize();
  public abstract Path input();
  public abstract Path output();
  public abstract String watermark();

  public static Watermark_WatermarkSettings_Builder builder() {
    return Watermark_WatermarkSettings_Builder.builder().fontSize(72);
  }
}
```

```java
private void init(Watermark.WatermarkSettings input) {
  fontSize = input.fontSize();
  input = input.input();
  output = input.output();
  watermark = input.watermark();
}
```

```
Watermark_WatermarkSettings_Builder.java:23: error: incompatible types: Path cannot be converted to WatermarkSettings
    input = input.input();
                       ^
1 error
```

**Comment by h908714124** [comment]
Oof. Nice catch. Should be resolved in 2.9.3.

**Comment by ben-manes** [comment]
Thank you 😊
